vm/systemSmartContracts/liquidStaking.go

```diff
@@ -182,14 +186,117 @@ func (l *liquidStaking) claimDelegatedPosition(args *vmcommon.ContractCallInput)
 		return vmcommon.UserError
 	}
 
+	numOfCalls := big.NewInt(0).SetBytes(args.Arguments[0]).Int64()
+	minNumArguments := numOfCalls*2 + 1
+	if int64(len(args.Arguments)) < minNumArguments {
+		l.eei.AddReturnMessage("invalid number of arguments")
+		return vmcommon.UserError
+	}
+	err := l.eei.UseGas(uint64(numOfCalls) * l.gasCost.MetaChainSystemSCsCost.LiquidStakingOps)
+	if err != nil {
+		l.eei.AddReturnMessage(err.Error())
+		return vmcommon.OutOfGas
+	}
+
+	listNonces := make([]uint64, 0)
+	listValues := make([]*big.Int, 0)
+	startIndex := int64(1)
+	for i := int64(0); i < numOfCalls; i++ {
+		callStartIndex := startIndex + i*2
+		nonce, valueToClaim, returnCode := l.claimOneDelegatedPosition(args.CallerAddr, args.Arguments[callStartIndex], args.Arguments[callStartIndex+1])
+		if returnCode != vmcommon.Ok {
+			return returnCode
+		}
+
+		listNonces = append(listNonces, nonce)
+		listValues = append(listValues, valueToClaim)
+	}
+
+	var additionalArgs [][]byte
+	if int64(len(args.Arguments)) > minNumArguments {
+		additionalArgs = args.Arguments[minNumArguments:]
+	}
+	err = l.sendNFTMultiTransfer(args.CallerAddr, listNonces, listValues, additionalArgs)
+	if err != nil {
+		l.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
 	return vmcommon.Ok
 }
 
+func (l *liquidStaking) claimOneDelegatedPosition(
+	callerAddr []byte,
+	destSCAddress []byte,
+	valueAsBytes []byte,
+) (uint64, *big.Int, vmcommon.ReturnCode) {
+	if len(destSCAddress) != len(l.liquidStakingSCAddress) || bytes.Equal(destSCAddress, l.liquidStakingSCAddress) {
```

Comment:
can we have other "invalid addresses" like the validator/staking SC and such?

Reply:
it does not matter. as those system scs will error for the given input. would work with len check only.